src/Hue.cpp

```diff
@@ -246,6 +246,34 @@ HueLight& Hue::getLight(int id)
     return lights.find(id)->second;
 }
 
+HueSensor& Hue::getSensor(int id)
+{
+    auto pos = sensors.find(id);
+    if (pos != sensors.end())
+    {
+        pos->second.refreshState();
+        return pos->second;
+    }
+    refreshState();
+    if (!state["sensors"].count(std::to_string(id)))
+    {
+        std::cerr << "Error in Hue getSensor(): sensor with id " << id << " is not valid\n";
+        throw HueException(CURRENT_FILE_INFO, "Sensor id is not valid");
+    }
+    // std::cout << state["sensors"][std::to_string(id)] << std::endl;
+    std::string type = state["sensors"][std::to_string(id)]["modelid"];
+    // std::cout << type << std::endl;
+    if (type == "RWL021" || type == "PHDL00" || type == "PHWA01")
```

Comment:
This should probably be added to HueDeviceTypes.cpp

Reply:
Should we really manage lights and sensors in one class? Does the types are shared between lights and sensors?